spec/forms/steps/case/charges_form_spec.rb

```diff
@@ -28,4 +28,38 @@
       end
     end
   end
+
+  describe '#offence_dates' do
+    it 'returns a list of offence date fieldset forms' do
+      charge_record.offence_dates << [
```

Comment:
I think if we can build the dates from the params is better as that's how this form is created in reality, through a post/put request to the controller, and passing the params.
Take a look at the `codefendants_form_spec` and the `let(:codefendants_attributes)`